Add Photoshop case study (#5717)
GoogleChrome/developer.chrome.com: site/en/blog/how-photoshop-solved-working-with-files-larger-than-can-fit-into-memory/index.md

---
layout: 'layouts/blog-post.njk'
title: 'How Photoshop solved working with files larger than can fit into memory'
subhead: >
  Learn how Adobe managed to let users edit even the biggest files on the web version of its iconic Photoshop app.

> date: 2023-03-21

Reviewer: ```suggestion
date: 2023-03-23
```

# updated: 2023-03-21

> hero: image/8WbTDNrhLsU0El80frMBGE4eMCD3/pYiISdhsJe3oEL871Dqp.png

Reviewer: I suggest adding a different thumbnail (using `thumbnail:`) here, if you look at the listing preview the image is not great. https://pr-5717-static-dot-dcc-staging.uc.r.appspot.com/blog/

Reviewer: That looks better, please merge this when you are ready.

Author: Thanks for the love for the detail. I didn't notice this because I only ever saw the page itself, not the overview.

alt: A part of Photoshop's user interface.

> authors:

Author: ```suggestion
thumbnail: image/8WbTDNrhLsU0El80frMBGE4eMCD3/tTlvAYtbWrg8xLqXdjua.png
authors:
```

Devices and platforms, be they desktop machines, mobile devices or browsers, all manage memory. Some are more generous than others in how much memory they make available to applications. The amount of memory also varies from device to device, as you know when you order a new computer or device and specify the amount of [Random-Access Memory](https://en.wikipedia.org/wiki/Random-access_memory) (RAM) desired. Many of these platforms also support [virtual memory](https://en.wikipedia.org/wiki/Virtual_memory), which allows an application to use more memory than is physically available. This support varies by operating system and runtime, as in the case of WebAssembly, may not be readily accessible or usable by applications. On top of that, modern virtual systems have upper limits that are easily exceeded by Photoshop requirements.

Ideally, applications would use as much memory as they need. This generally allows them to provide the best performance to their users. However, if they use too much memory, they may be penalized by the runtime platform or may run out of memory, resulting in failures.

> As a historical note, the original problem that Photoshop needed to solve was editing print resolution files on early versions of macOS, as low as 1MB for the OS and all applications. A 300 dpi full page image in CMYK is approximately 32MB uncompressed.

Reviewer: ```suggestion
As a historical note, the original problem that Photoshop needed to solve was editing print resolution files on early versions of macOS, as low as 1&nbsp;MB for the OS and all applications. A 300&nbsp;dpi full page image in CMYK is approximately 32&nbsp;MB uncompressed.
```

Reviewer: As per styleguide https://developers.google.com/style/units-of-measure

---
layout: 'layouts/blog-post.njk'
title: 'How Photoshop solved working with files larger than can fit into memory'
subhead: >
  Learn how Adobe managed to let users edit even the biggest files on the web version of its iconic Photoshop app.
date: 2023-03-23
# updated: 2023-03-21
hero: image/8WbTDNrhLsU0El80frMBGE4eMCD3/pYiISdhsJe3oEL871Dqp.png
alt: A part of Photoshop's user interface.
thumbnail: image/8WbTDNrhLsU0El80frMBGE4eMCD3/tTlvAYtbWrg8xLqXdjua.png
authors:
  - nabeelalshamma
  - thomassteiner
tags:
  - capabilities
  - case-study
  - fugu-case-study
---

## Introduction

> 

Author: ```suggestion

(This article is also available in form of a video.)

{% YouTube id="Y40vMQap9fs", startTime="695" %}
```

---
layout: 'layouts/blog-post.njk'
title: 'How Photoshop solved working with files larger than can fit into memory'
subhead: >
  Learn how Adobe managed to let users edit even the biggest files on the web version of its iconic Photoshop app.
date: 2023-03-23
# updated: 2023-03-21
hero: image/8WbTDNrhLsU0El80frMBGE4eMCD3/pYiISdhsJe3oEL871Dqp.png
alt: A part of Photoshop's user interface.
thumbnail: image/8WbTDNrhLsU0El80frMBGE4eMCD3/tTlvAYtbWrg8xLqXdjua.png
authors:
  - nabeelalshamma
  - thomassteiner
tags:
  - capabilities
  - case-study
  - fugu-case-study
---

## Introduction

(This article is also available in form of a video.)

> {% YouTube id="Y40vMQap9fs", startTime="695" %}

Author: ```suggestion
{% YouTube id="Y40vMQap9fs", startTime="695" %}

```

---
layout: 'layouts/blog-post.njk'
title: 'How Photoshop solved working with files larger than can fit into memory'
subhead: >
  Learn how Adobe managed to let users edit even the biggest files on the web version of its iconic Photoshop app.

> date: 2023-03-23
> # updated: 2023-03-21

Author: ```suggestion
date: 2023-06-05
# updated: 2023-06-05
```
